lib/src/rsa.nr

```diff
@@ -48,6 +114,113 @@ fn compare_signature_sha256<let N: u32>(padded_sha256_hash: [u8; N], msg_hash: [
 }
 impl<BN, BNInstance, let NumBytes: u32> RSA<BN, BNInstance, NumBytes>  where BN: BigNumTrait, BNInstance: BigNumInstanceTrait<BN> {
     /**
+    * @brief Verify an RSA signature generated via the PSS signature scheme.
+    * @details `key_size` is the size of the RSA modulus in bits and is required to correctly decode the signature.
+    *
+    * @note We assume the public key exponent `e` is 65537
+    **/
+    pub fn verify_sha256_pss(
+        _: Self,
```

Comment:
As far as I know, this is required for methods of a Struct. At least the compiler won't let me do without it

Reply:
You should be able to remove it, but the method may need to be moved to a separate impl since the generics on the impl should no longer apply either. Since there'd be no self type you'd also have to call it via the fully qualified syntax.

Reply:
These changes have been applied with which commit of the Noir repo exactly?

Reply:
I don't really see how to use it in this case though since the generics are at the level of the impl and not the function itself. So if I try something like this:
```rust
RSA::<BN2048, BNInst2048, 256>::verify_sha256_pss(BNInstance, sha256_hash, signature, 2048);
```
or
```rust
type RSA2048 = RSA<BN2048, BNInst2048, 256>;
RSA2048::verify_sha256_pss(BNInstance, sha256_hash, signature, 2048)
```
None of the two works. Also, I believe the point of how this lib has been developed is to abstract away all the generics so at the end of the day you can just use something like RSA2048 with everything already predefined in the library. For example, the original PKCS#1 signature verification:
```rust
let rsa: RSA2048 = RSA {};
assert(rsa.verify_sha256_pkcs1v15(BNInstance, sha256_hash, signature));
```